core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -285,9 +285,10 @@ private[spark] object Utils extends Logging {
    */
   def createDirectory(dir: File): Boolean = {
     try {
-      // This sporadically fails - not sure why ... !dir.exists() && !dir.mkdirs()
-      // So attempting to create and then check if directory was created or not.
-      dir.mkdirs()
+      // SPARK-35907
+      // This could throw more meaningful exception information if directory creation failed.
```

Comment:
nit: "SPARK-35907: This could..."

Reply:
File line length exceeds 100 characters. I was going to do that.

Reply:
I kind of feel the comments aren't necessary? The comment explains why we switched from `File#mkdirs()` to `Files.createDirectories()`, but IMO a comment should explain something that is confusing or unexpected, and if you just come across `Files.createDirectories()`, there's nothing strange about that.